modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/listk.kt

```diff
@@ -279,3 +291,17 @@ interface ListKAlternative : Alternative<ForListK>, ListKApplicative {
   override fun <A> Kind<ForListK, A>.orElse(b: Kind<ForListK, A>): Kind<ForListK, A> =
     (this.fix() + b.fix()).k()
 }
+
+@extension
+interface ListKSemialign : Semialign<ForListK>, ListKFunctor {
+  override fun <A, B> align(
+    left: Kind<ForListK, A>,
+    right: Kind<ForListK, B>
+  ): Kind<ForListK, Ior<A, B>> = alignRec(left.fix(), right.fix()).k()
+
+  private fun <X, Y> alignRec(ls: List<X>, rs: List<Y>): List<Ior<X, Y>> = when {
+    ls.isEmpty() -> rs.map { it.rightIor() }
+    rs.isEmpty() -> ls.map { it.leftIor() }
+    else -> listOf(Ior.Both(ls.first(), rs.first())).listPlus(alignRec(ls.drop(1), rs.drop(1)))
+  }
+}
```

Comment:
Why not:
```kotlin
fun align() = when {
    ls.isEmpty() -> rs.map { it.rightIor() }
    rs.isEmpty() -> ls.map { it.leftIor() }
    else -> listOf(Ior.Both(ls.first(), rs.first())).listPlus(alignRec(ls.drop(1), rs.drop(1)))
  }
```
directly? :see_no_evil: Sorry if my comments have been a bit misleading on that

Reply:
i tried that but could not make it work initially. issue is that you have fix() ls and rs first, also listPlus is more complicated as i then try to add Kind<ForListK> to List. i will look into it again and see if i can find a way.

Reply:
ah yeah that's always fun. makes it a bit uglier but you can do it either by calling fix every time or stick a `(ls.fix() toT rs.fix()).let { (ls, rs) -> }` to get it done once before. Does not matter to me and if you think the private function is fine then just keep it. Up to you :)